api/http_lifecycle.go

```diff
@@ -21,28 +21,23 @@ var (
 	ErrNoOperationGiven = errors.New("lfs/api: no operation provided in schema")
 )
 
-// EndpointSource is an interface which encapsulates the behavior of returning
-// `config.Endpoint`s based on a particular operation.
-type EndpointSource interface {
-	// Endpoint returns the `config.Endpoint` assosciated with a given
-	// operation.
-	Endpoint(operation string) config.Endpoint
-}
-
 // HttpLifecycle serves as the default implementation of the Lifecycle interface
 // for HTTP requests. Internally, it leverages the *http.Client type to execute
 // HTTP requests against a root *url.URL, as given in `NewHttpLifecycle`.
 type HttpLifecycle struct {
-	endpoints EndpointSource
+	cfg *config.Configuration
 }
 
 var _ Lifecycle = new(HttpLifecycle)
 
 // NewHttpLifecycle initializes a new instance of the *HttpLifecycle type with a
 // new *http.Client, and the given root (see above).
-func NewHttpLifecycle(endpoints EndpointSource) *HttpLifecycle {
+func NewHttpLifecycle(cfg *config.Configuration) *HttpLifecycle {
```

Comment:
Do you think it'd be worth documenting the default behavior if `nil` is given here?